src/interface/metadata.hpp

```diff
@@ -181,6 +181,13 @@ class Metadata {
     SetMultiple(bits);
   }
 
+  /// returns a metadata with bits and shape set
+  explicit Metadata(const std::vector<MetadataFlag> &bits, const std::vector<int> &shape,
+                    const std::vector<std::string> component_labels)
+      : shape_(shape), sparse_id_(-1), component_labels_(component_labels) {
```

Comment:
Are there any consistency rules between the `shape` and the number of `component_labels` that we should/could enforce?

Reply:
Not any simple rules right now. Currently, the `yt` interface would accept any number of `component_labels` as long as it could reshape the raw array into right `nlabels*nz*ny*nx` shape. Since we're not sure at the moment how downstream codes will arrange or label their components, I'm not sure there's a simple rule that we can impose at the moment.